pkg/logs/input/journald/tailer_test.go

```diff
@@ -148,5 +150,67 @@ func TestApplicationNameShouldBeDockerForContainerEntries(t *testing.T) {
 				sdjournal.SD_JOURNAL_FIELD_COMM:              "foo.sh",
 				containerIDKey:                               "bar",
 			},
-		}))
+		}, []string{}))
+}
+
+func TestApplicationNameShouldBeShortImageForContainerEntries(t *testing.T) {
+	containerID := "bar"
+
+	source := config.NewLogSource("", &config.LogsConfig{ContainerMode: true})
+	tailer := NewTailer(source, nil)
+
+	assert.Equal(t, "testImage", tailer.getApplicationName(
+		&sdjournal.JournalEntry{
+			Fields: map[string]string{
+				sdjournal.SD_JOURNAL_FIELD_SYSLOG_IDENTIFIER: "foo",
+				sdjournal.SD_JOURNAL_FIELD_SYSTEMD_UNIT:      "foo.service",
+				sdjournal.SD_JOURNAL_FIELD_COMM:              "foo.sh",
+				containerIDKey:                               containerID,
+			},
+		}, []string{"short_image:testImage"}))
+
+	// Verify we have the value in our cache
+	_, hit := cache.Cache.Get(getImageCacheKey(containerID))
+	assert.True(t, hit)
+}
+
+func TestApplicationNameShouldBeDockerWhenTagNotFound(t *testing.T) {
+	containerID := "bar2"
+
+	source := config.NewLogSource("", &config.LogsConfig{ContainerMode: true})
+	tailer := NewTailer(source, nil)
+
+	assert.Equal(t, "docker", tailer.getApplicationName(
+		&sdjournal.JournalEntry{
+			Fields: map[string]string{
+				sdjournal.SD_JOURNAL_FIELD_SYSLOG_IDENTIFIER: "foo",
+				sdjournal.SD_JOURNAL_FIELD_SYSTEMD_UNIT:      "foo.service",
+				sdjournal.SD_JOURNAL_FIELD_COMM:              "foo.sh",
+				containerIDKey:                               containerID,
+			},
+		}, []string{"not_short_image:testImage"}))
+
+	// Verify we don't have value in our cache
+	_, hit := cache.Cache.Get(getImageCacheKey(containerID))
+	assert.False(t, hit)
+}
+
+func TestWrongTypeFromCache(t *testing.T) {
+	containerID := "bar3"
+
+	// Store wrong type in cache, verify we ignore the value
+	cache.Cache.Set(getImageCacheKey(containerID), 10, 30*time.Second)
+
+	source := config.NewLogSource("", &config.LogsConfig{ContainerMode: true})
+	tailer := NewTailer(source, nil)
+
+	assert.Equal(t, "docker", tailer.getApplicationName(
+		&sdjournal.JournalEntry{
+			Fields: map[string]string{
+				sdjournal.SD_JOURNAL_FIELD_SYSLOG_IDENTIFIER: "foo",
+				sdjournal.SD_JOURNAL_FIELD_SYSTEMD_UNIT:      "foo.service",
+				sdjournal.SD_JOURNAL_FIELD_COMM:              "foo.sh",
+				containerIDKey:                               containerID,
+			},
+		}, []string{"not_short_image:testImage"}))
```

Comment:
even though it won't change the result, we probably want to use the correct tag here, e.g. `short_image:testImage `

Reply:
Changed